### PR TITLE
chore: use Link component and add aria-label

### LINKED ISSUE #1943: [Refactor] Improve SEO Optimization and accessibility
### File Name

Multiple Components

### Reason for Refactoring the Code

- Convert `<a>` anchor tag to `<Link>` component.

Reference Docs: https://nextjs.org/docs/pages/api-reference/components/link

- Add `aria-label` attribute to every link and button in the codebase

The `aria-label` attribute is an essential accessibility feature in HTML. It is used to provide a descriptive label for elements that might not have visible text or when the existing text content is not sufficient to convey the element's purpose to users of assistive technologies (e.g., screen readers).

In short, adding `aria-label` helps provide context to users who rely on assistive technologies.

### Record

- [X] I have checked the existing [issues](https://github.com/priyankarpal/ProjectsHut/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/priyankarpal/ProjectsHut/blob/main/contributing.md)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/priyankarpal/ProjectsHut/blob/main/CODE_OF_CONDUCT.md)
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Anmol-Baranwal**
## Related Issue

Resolves #1943 

## Description

- convert anchor tag to `<Link>` 
- add `aria-label` to links & buttons

### CODE REVIEW

File: components/Navbar.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ const Navbar: NextPage = () => {
               {navLinks.map((navLink) => (
                 <li key={navLink.path}>
                   {navLink.external ? (
-                    <a
+                    <Link
```
Comment (priyankarpal) [inline components/Navbar.tsx:58]: add `aria-label="desktop navbar link"`

```suggestion
                    <Link aria-label="desktop navbar link"
```

File: components/projects/ProjectList.tsx [hunk 2]
```diff
@@ -289,7 +289,7 @@ function ProjectList() {
                                             value={window.location.href}
                                             disabled
                                         />
-                                        <button onClick={() => copyText()}>
+                                        <button onClick={() => copyText()} aria-label="copy text">
```
Comment (priyankarpal) [inline components/projects/ProjectList.tsx:292]:
```suggestion
                                        <button onClick={() => copyText()} aria-label="copy button">
```

**Review by priyankarpal — CHANGES_REQUESTED** [review]

**Comment by Anmol-Baranwal** [comment]
Done.

**Review by priyankarpal — APPROVED** [review]
Looks Good 🎉

**Pull request merged**